### ISSUE TITLE
UI issues faced by a new user

### ISSUE THREAD

**Issue by gphilip** [zeapo/Android-Password-Store#62]
I installed Android-Password-Store today and tried to set it up to work with my Pass store, but I have failed so far. I want to tell you what didn't work, in the hope that it will help you improve the software.

1. The main issue I faced is that there is absolutely no feedback from the UI.  This is the most frustrating thing when things do *not* working as intended, because one doesn't have a clue as to at which step the process failed. For instance:

   I use the "import ssh-key" button(?) to import an ssh key, but I have no clue whether the key is successfully imported or not. There is no message either confirming or denying that the import succeeded. So, when I am unable to clone my repository later, I don't know if this is because the ssh key import failed, or not.

  Even a way to look at some kind of a text log to figure out if importing the ssh key (or other steps) succeeded, would be of great help in fixing things. 

2. I used the "Edit git server settings" page/button/ (what exactly is the name of this UI element?) to add my git server settings. It turns out that if I enter anything at all in the "Username" field, then when I try later to pull from remote I get the following error message from jGit:

  "/nPull on repository without HEAD currently not supported"

When I first got this message I didn't know if this was because the ssh key import failed (see 1 above), but then I figured out by trial and error that the software gives this error only when  the Username field is nonempty. Also, this happens locally, without any attempt at connecting to the git server.

Now my repository is at bitbucket.org, and as per their documentation (https://confluence.atlassian.com/display/BITBUCKET/Use+the+SSH+protocol+with+Bitbucket), for accessing a repository via ssh one should make the URL like so: 

ssh://git@bitbucket.org/accountname/reponame.git

I could not figure out how to achieve this exact form for the "Resulting URL" while leaving the Username part empty. The best I could come up with was to set:

Username: (empty; forced by the jGit error message mentioned above)
Server URL: git@bitbucket.org
Repo path: accountname/reponame.git

But then this gives the following "Resulting URL":

@git@bitbucket.org:accountname/reponame.git

Note the extra "@" at the beginning. This definitely doesn't look kosher, but when I am unable to pull from remote, I don't know if this extra "@" is the reason, because there is no feedback or log to look at.

(This is probably the same as issue #61 .)

If I edit the "Resulting URL" field so that it matches the required form, then the software fills in the "Username" field with the word "git" (which *is* the correct username as per bitbucket's documentation), and then I get the jGit error which I mentioned above. 

3. After setting things up like this, when I try to pull from remote, I get, in turn:

  a) A prompt for the passphrase for my SSH key. Whatever I do here has no bearing on what eventually happens: I have tried leaving it empty, typing in the passphrase for my ssh key, and typing in junk, and in all cases the final result (after the next step) is the same.

  b) A prompt that I have to "set the information about the server before synchronizing with the server". The only possible way to progress is to choose "Nah ... later" (Otherwise it takes me back to the server settings screen where I don't know what else to do.), and when I choose this then I get one big blank page saying "Password Store" at the top and nothing much else. 

The software does *not* try to connect to any server whatsoever (I run a firewall, and I would get a prompt from there if the software tried to make a connection). It also does not tell me if

 (i) it is trying to connect,

 (ii) something is wrong

 (iii) or anything else;

it just sits there. 

I am sorry for this rambling report, but I didn't see anything else resembling this (except for issue #61 ) so I thought I would add an issue and mention these things.

 
Edit: minor grammar fix.

**Comment by zeapo** [comment]
Thank you for the detailed report, especially the UI issues.

For the #61, the user had this issue when he tried to log using a username/password on the ssh protocol. Is this your case too?

**Comment by gphilip** [comment]
>For the #61, the user had this issue when he tried to log using a username/password on the ssh protocol. Is this your case too?

No, I have been trying to log in using an ssh keypair.

**Comment by zeapo** [comment]
Are you using the 1.2.0.11 (PlayStore) or 1.2.0.10 (F-Droid) ?

On Sun, Jan 4, 2015 at 1:31 PM, gphilip <notifications@github.com> wrote:

> For the #61 <https://github.com/zeapo/Android-Password-Store/issues/61>,
> the user had this issue when he tried to log using a username/password on
> the ssh protocol. Is this your case too?
>
> No, I have been trying to log in using an ssh keypair.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeapo/Android-Password-Store/issues/62#issuecomment-68631289>
> .
>



-- 
Mohamed Zenadi

mail: mohamed@zenadi.com
webpage: http://zenadi.com

**Comment by gphilip** [comment]
I am using the F-droid one.

**Comment by smblott-github** [comment]
Off topic, but...

> Are you using the 1.2.0.11 (PlayStore) or 1.2.0.10 (F-Droid) ?

But you also say...

> F-Droid (the prefered way)
> Play Store (always lags behind)

**Comment by gphilip** [comment]
>But you also say...

>>F-Droid (the prefered way)
 >>Play Store (always lags behind)


This is exactly the reason why I chose the F-droid version.

**Comment by zeapo** [comment]
There was a glitch in the matrix :)

On Sun, Jan 4, 2015 at 2:13 PM, gphilip <notifications@github.com> wrote:

> But you also say...
>
> F-Droid (the prefered way)
> Play Store (always lags behind)
>
>  This is exactly the reason why I chose the F-droid version.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeapo/Android-Password-Store/issues/62#issuecomment-68632426>
> .
>



-- 
Mohamed Zenadi

mail: mohamed@zenadi.com
webpage: http://zenadi.com

**Comment by zeapo** [comment]
> Now my repository is at bitbucket.org, and as per their documentation (https://confluence.atlassian.com/display/BITBUCKET/Use+the+SSH+protocol+with+Bitbucket), for accessing a repository via ssh one should make the URL like so:
> ssh://git@bitbucket.org/accountname/reponame.git
> 
> I could not figure out how to achieve this exact form for the "Resulting URL" while leaving the Username part empty. The best I could come up with was to set:

> Username: (empty; forced by the jGit error message mentioned above)
Server URL: git@bitbucket.org
Repo path: accountname/reponame.git

> But then this gives the following "Resulting URL":

> @git@bitbucket.org:accountname/reponame.git

> Note the extra "@" at the beginning. This definitely doesn't look kosher, but when I am unable to pull from remote, I don't know if this extra "@" is the reason, because there is no feedback or log to look at.

> (This is probably the same as issue #61 .)

> If I edit the "Resulting URL" field so that it matches the required form, then the software fills in the "Username" field with the word "git" (which is the correct username as per bitbucket's documentation), and then I get the jGit error which I mentioned above.

The username in the application corresponds to the ssh username which in many cases is `git`. A user not used to git might think that it corresponds to his username. This is indeed an issue, If you have a way to solve it, it would be more than welcome :)

**Comment by gphilip** [comment]
> The username in the application corresponds to the ssh username which in many cases is git. A user not used to git might think that it corresponds to his username. This is indeed an issue, If you have a way to solve it, it would be more than welcome :)

What I found was that if I give *any* nonempty string as "username", then I would get the jGit error. This happens even when I give "git" as the username. The only way for the software to not error out with the jGit error is to leave this field empty.

This is about the version which was on F-Droid. I will update it to the latest tomorrow (when you said it would be up-to-date), and check again.

Thanks for taking this up!

**Comment by zeapo** [comment]
tomorrow's version wont solve this problem, AFAIK.

Could you get a log using logcat when you get this jgit error? I am unable
to reproduce this error.

On Sun, Jan 4, 2015 at 6:03 PM, gphilip <notifications@github.com> wrote:

> The username in the application corresponds to the ssh username which in
> many cases is git. A user not used to git might think that it corresponds
> to his username. This is indeed an issue, If you have a way to solve it, it
> would be more than welcome :)
>
> What I found was that if I give *any* nonempty string as "username", then
> I would get the jGit error. This happens even when I give "git" as the
> username. The only way for the software to not error out with the jGit
> error is to leave this field empty.
>
> This is about the version which was on F-Droid. I will update it to the
> latest tomorrow (when you said it would be up-to-date), and check again.
>
> Thanks for taking this up!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeapo/Android-Password-Store/issues/62#issuecomment-68639988>
> .
>



-- 
Mohamed Zenadi

mail: mohamed@zenadi.com
webpage: http://zenadi.com

**Comment by zeapo** [comment]
Show the import to the user implemented in : https://github.com/zeapo/Android-Password-Store/commit/1c2b000847b73c482154f5ccc15cf2cfb9865c4d#diff-b697c23b5ea41540b8ed7ff9937a7a8cR157

**Comment by zeapo** [comment]
@gphilip can you check that some of the issues are solved in https://f-droid.org/repository/browse/?fdid=com.zeapo.pwdstore (v1.2.0.12)
Could you also, make a checklist in your first issue so that we can keep track of what is lacking. Thanks :)

**Comment by gphilip** [comment]
@zeapo , sorry for the delay in responding, I had only intermittent access to the internet these last few days.

I updated to the latest version in F-Droid yesterday, and now the software works absolutely as expected:
1. The ssh key import tells me that it succeeded.
2. Entering git login details works as expected: Thus, for instance, setting "git" as the username (as it should be) for my account on bitbucket works (as it should).
3. I was able to pull from the repository.
4. After installing OpenKeyChain and importing my gpg key into it, I am now able to view passwords stored in the git repository. 

I am now able to use the software as I want to use it, and all the issues I reported are resolved. 

Thanks a lot for the quick fixes!!
